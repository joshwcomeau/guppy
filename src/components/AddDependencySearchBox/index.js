// @flow
export { default } from './AddDependencySearchBox';
