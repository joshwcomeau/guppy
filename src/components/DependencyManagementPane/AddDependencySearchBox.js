// @flow
/**
 * This component uses Algolia's `react-instantsearch`.
 * Unfortunately, styling these components is tricky, and doesn't play nicely
 * with styled-components. Just gonna override the provided global styles.
 */
import React, { Component } from 'react';
import styled, { injectGlobal } from 'styled-components';
import { SearchBox } from 'react-instantsearch/dom';

import { COLORS } from '../../constants';

type Props = {
  onChange?: (ev: SyntheticEvent<*>) => void,
};

class AddDependencySearchBox extends Component<Props> {
  render() {
    return (
      <Wrapper>
        <SearchBox
          onChange={this.props.onChange}
          autoFocus={true}
          placeholder="eg. redux, react-router..."
        />
      </Wrapper>
    );
  }
}

const Wrapper = styled.div`
  padding-top: 15px;
  width: 100%;
`;

injectGlobal`
  .ais-SearchBox {
    /* width: 100%; */
  }
  .ais-SearchBox-submit, .ais-SearchBox-reset {
    display: none;
  }
  .ais-SearchBox-input {
    width: 100%;
    padding: 8px 0px;
    background: transparent;
    border: none;
    border-bottom: 2px solid rgba(255, 255, 255, 0.75);
    color: ${COLORS.lightBackground};
    border-radius: 0px;
    outline: none;
    font-size: 21px;

    &::placeholder {
      color: rgba(255, 255, 255, 0.35);
    }

    &:focus {
      border-bottom: 2px solid ${COLORS.lightBackground};

    }
  }
`;

export default AddDependencySearchBox;
