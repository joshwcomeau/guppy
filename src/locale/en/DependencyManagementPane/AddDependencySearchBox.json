{
  "dependencyManagementPane.addDependencySearchBox.main.placeholder": "eg. redux, react-router..."
}