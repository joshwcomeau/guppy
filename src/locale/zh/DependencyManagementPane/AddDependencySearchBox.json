{
  "dependencyManagementPane.addDependencySearchBox.main.placeholder": "例如redux、react-router..."
}