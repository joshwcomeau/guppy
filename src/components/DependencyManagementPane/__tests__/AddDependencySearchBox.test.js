import React from 'react';
import { shallow } from 'enzyme';

import AddDependencySearchBox from '../AddDependencySearchBox';

describe('AddDependencySearchBox component', () => {
  const wrapper = shallow(<AddDependencySearchBox />);

  it('should render SearchBox', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
