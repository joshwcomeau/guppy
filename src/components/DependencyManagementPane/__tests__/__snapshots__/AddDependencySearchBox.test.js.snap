// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDependencySearchBox component should render SearchBox 1`] = `
<styled.div>
  <AlgoliaSearchBox(Translatable(SearchBox))
    autoFocus={true}
    placeholder="eg. redux, react-router..."
  />
</styled.div>
`;
